### PR TITLE
Update the dependencies of the Java Binding

### PR DESCRIPTION
**Pull request by yvanzo**
No functional change, just avoid to start lagging behind updates again.

### CODE REVIEW

**Review by atj — COMMENTED** [review]
LGTM but a bit beyond my level of expertise, so a review by lucifer and/or bitmap would be good!

File: brainz-mmd2-jaxb/src/main/java/org/musicbrainz/mmd2/AliasList.java [hunk 1]
```diff
@@ -56,22 +56,23 @@ public class AliasList {
...
      * Gets the value of the alias property.
      * 
-     * <p>
-     * This accessor method returns a reference to the live list,
+     * <p>This accessor method returns a reference to the live list,
      * not a snapshot. Therefore any modification you make to the
-     * returned list will be present inside the Jakarta XML Binding object.
-     * This is why there is not a {@code set} method for the alias property.
+     * returned list will be present inside the JAXB object.
+     * This is why there is not a <CODE>set</CODE> method for the alias property.</p>
```
Comment (amCap1712) [inline brainz-mmd2-jaxb/src/main/java/org/musicbrainz/mmd2/AliasList.java:62]: Due to these comment updates, I have a slight doubt if the correct Jaxb plugin version was used during the update of these classes. Jakarta XML Binding is the new name of the artifact so it seems unlikely to me that the plugin authors would revert to the older JAXB terminology. But of course this is only a hunch and I could be wrong. I didn't find anything obvious from the release notes of latest 2-3 releases of the plugin.

I would suggest regenerating the schema classes after invalidating all caches locally and see if something changes.

Comment (yvanzo) [reply brainz-mmd2-jaxb/src/main/java/org/musicbrainz/mmd2/AliasList.java:62]: OK, I tried the following but it made no difference:
```bash
sudo rm -fr docker/brainz-mmd2-jaxb/bash_history/ docker/brainz-mmd2-jaxb/maven_cache/
docker compose --project-directory=docker/brainz-mmd2-jaxb build --no-cache builder
docker compose --project-directory=docker/brainz-mmd2-jaxb run --build --rm builder
```

**Review by amCap1712 — APPROVED** [review]
LGTM, Thanks!

File: brainz-mmd2-jaxb/src/main/java/org/musicbrainz/mmd2/AliasList.java (hunk 1 above)
Comment (yvanzo) [reply brainz-mmd2-jaxb/src/main/java/org/musicbrainz/mmd2/AliasList.java]: I tried different [versions of `jaxb-maven-plugin`](https://central.sonatype.com/artifact/org.jvnet.jaxb/jaxb-maven-plugin/versions) (4.0.6, 4.0.3, 4.0.2) until it started to mention Jakarta again with version 4.0.0 (which is the version suggested in the [documentation](https://github.com/highsource/jaxb-tools?tab=readme-ov-file#quick-start)), thus sticking with this version for now, and reporting a potential issue upstream.

**Pull request merged**